tests/tests/test_decommission.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2021 Northern.tech AS
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import pytest
+
+from ..common_setup import standard_setup_one_client
+from ..MenderAPI import authentication, devauth
+from .mendertesting import MenderTesting
+
+
+class TestDecommission(MenderTesting):
+    """Tests the decommission and re authentication functionality"""
+
+    def test_decommission(self, standard_setup_one_client):
+        mender_device = standard_setup_one_client.device
+
+        # accept the device
+        devauth.accept_devices(1)
+        # list of authorized devices
+        devices = list(
+            set([device["id"] for device in devauth.get_devices_status("accepted")])
+        )
+        # check that there is a authorized device
+        assert 1 == len(devices)
+        # check from client side that the authorization was successful
+        journal_info = self.get_device_journal(mender_device)
+        assert "Server authorization successful" in journal_info
+
+        # decommission the device
+        devauth.decommission(devices[0])
+        devices = list(
+            set([device["id"] for device in devauth.get_devices_status("accepted")])
+        )
+        # check that there is no authorized devices after decommission
+        assert 0 == len(devices)
+        old_journal_info = journal_info
+        journal_info = self.get_device_journal(mender_device)
+        # check on the client side that no new authorization has happened
+        assert "Server authorization successful" not in self.remove_old_journal_info(
+            old_journal_info, journal_info
+        )
```

Comment:
@lluiscampos Are assertions like this with a function call a decent solution, or is it a bit messy?

Reply:
Assert on a function return value is okey.

It looks better now, but we are still copy-pasting the "Server authorization..." text. If the client ever changes that log line, I would expect to come to this test and modify only one line. For that, what I meant was to have a function like `def verify_device_authorized(expect_authorized=True)` so that this assert happens inside.

Reply:
Ah, that was clever! I will fix it now :+1:

Reply:
Ah, that was clever, I'm fixing it now :+1: